bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/discovery/SunSynkAccountDiscoveryService.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.sunsynk.internal.discovery;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.openhab.binding.sunsynk.internal.SunSynkBindingConstants;
+import org.openhab.binding.sunsynk.internal.classes.Inverter;
+import org.openhab.binding.sunsynk.internal.handler.SunSynkAccountHandler;
+import org.openhab.binding.sunsynk.internal.handler.SunSynkHandlerFactory;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SunSynkAccountDiscoveryService} is responsible for starting the discovery procedure
+ * that connects to SunSynk Web and imports all registered inverters.
+ * 
+ * @author Lee Charlton - Initial contribution
+ */
+
```

Comment:
Annotate with nonnullbydefault

Reply:
Not sure I can do this, any help apricated. 
`Unsafe interpretation of method return type as '@NonNull' based on substitution 'T=@NonNull Details'. Declaring type 'Gson' doesn't seem to be designed with null type annotations in mind`
gson.fromJson is only attempted in a try catch.

Reply:
Null annotations and compiler are not always best friends. I think something like this should work:
```
@Nullable
Details output = gson.fromJson(response, Details.class);
```

Reply:
Have now tried this, but get a Compilation failure.
Sorry I  don't  yet understand enough these annotations, as I annotate, I find they just cascade out.
In SunSynkAccountHandler:
`Compilation failure:       
[ERROR] C:\Users\Lee\development\openhab-addons\bundles\org.openhab.binding.sunsynk\src\main\java\org\openhab\binding\sunsynk\internal\handler\SunSynkAccountHandler.java:[97,20] Null type mismatch (type annotations): required 'org.openhab.binding.sunsynk.internal.api.dto.@NonNull Details' but this expression has type 'org.openhab.binding.sunsynk.internal.api.dto.@Nullable Details'`

Unsafe interpretation of method return type as '@NonNull' based on substitution 'T=@NonNull Client'. Declaring type 'Gson' doesn't seem to be designed with null type annotations in mindJava(16778197)

Also get
In SunSynkAccountDiscovery Service using @NonNullByDefault

This line I think part  of the Skelton code
```java
 private ScheduledFuture<?> scanTask;
```

`Compilation failure: Compilation failure: 
[ERROR] C:\Users\Lee\development\openhab-addons\bundles\org.openhab.binding.sunsynk\src\main\java\org\openhab\binding\sunsynk\internal\discovery\SunSynkAccountDiscoveryService.java:[50,12] The @NonNull field scanTask may not have been initialized`

Reply:
Sorted